### ISSUE TITLE
intermittently fails to download driver when running in parallel

### ISSUE THREAD

**Issue by damonsmith** [bonigarcia/webdrivermanager#186]
Hi, I've got a set of tests that run in parallel, in 8 browsers at the same time. It uses TestNG and Surefire to do the parallel tests. They run on an ubuntu VM so each time a test run starts they need to download the chromedriver binaries but intermittently they just report that the binary is not found.

The logs I get are:
java.lang.IllegalStateException: The path to the driver executable must be set by the webdriver.chrome.driver system property; for more information, see https://github.com/SeleniumHQ/selenium/wiki/ChromeDriver. The latest version can be downloaded from http://chromedriver.storage.googleapis.com/index.html

**Comment by bonigarcia** [comment]
I would need to reproduce the error. Could you please provide a running example in which this behavior can be observed?

**Comment by bonigarcia** [comment]
@damonsmith any update on this?

**Comment by damonsmith** [comment]
Sorry i haven't had a chance, I'll try to get into it this week.

**Comment by dimkin-eu** [comment]
having same issue. it appears after we bump selenium 3.53 > 3.10 
Selenium itself shouldnt affect, but there is such issue https://github.com/seleniumhq/selenium-google-code-issue-archive/issues/5245
so maybe some issues with `org.apache.poi`/`xml-apis` ?

**Comment by bonigarcia** [comment]
@damonsmith Could you provide some example to reproduce it?

**Comment by bonigarcia** [comment]
@dimkin-eu Could you provide some example to reproduce it?

**Comment by dimkin-eu** [comment]
i will try to create some small scenario. 
but setup is same - jenkins job with many ( 20 in our case) parallel jobs
is there possibility to have more stacktrace from wdm ?

**Comment by bonigarcia** [comment]
Thank you @dimkin-eu.

Regarding logs, yes, wdm is logging additional info in the `TRACE` level. You simply need to change your logging configuration to get those traces. For instance, in the case of using logback, it would be as follows:

`<logger name="io.github.bonigarcia" level="TRACE" />`

**Comment by dimkin-eu** [comment]
due bug in chromedriver we are using 2.33
`ChromeDriverManager.getInstance().version(CHROME_DRIVER_VERSION).setup();`

```
10:02:05.818 TRACE[main] [io.github.bonigarcia.wdm.WdmConfig] Property key wdm.architecture not found, using default value
10:02:05.907 TRACE[main] [io.github.bonigarcia.wdm.WdmConfig] Property key wdm.os not found, using default value
10:02:05.908 TRACE[main] [io.github.bonigarcia.wdm.WdmConfig] Property key wdm.proxy not found, using default value
10:02:05.908 TRACE[main] [io.github.bonigarcia.wdm.WdmConfig] Property key wdm.proxyUser not found, using default value
10:02:05.908 TRACE[main] [io.github.bonigarcia.wdm.WdmConfig] Property key wdm.proxyPass not found, using default value
10:02:05.908 TRACE[main] [io.github.bonigarcia.wdm.WdmConfig] Property key wdm.useMirror not found, using default value
10:02:06.220 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] >> Managing [chromedriver] arch=64 version=2.33 getLatest=false cache=false
10:02:06.221 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Target path /home/jenkins/.m2/repository/webdriver
10:02:06.221 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking if chromedriver 2.33 (64 bits) exists in cache /home/jenkins/.m2/repository/webdriver
10:02:06.230 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/geckodriver/linux64/0.16.1/geckodriver
10:02:06.230 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver
10:02:06.230 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.30/chromedriver
10:02:06.230 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.29/chromedriver
10:02:06.231 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.28/chromedriver
10:02:06.231 INFO [main] [io.github.bonigarcia.wdm.WebDriverManager] Reading https://chromedriver.storage.googleapis.com/ to seek [chromedriver]
10:02:06.694 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] All URLs:
```
blablabla
```
10:02:06.695 DEBUG[main] [io.github.bonigarcia.wdm.WebDriverManager] Using [chromedriver] 2.33
10:02:06.695 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Candidate URLs: [https://chromedriver.storage.googleapis.com/2.33/chromedriver_win32.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_mac64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux32.zip]
10:02:06.696 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs before filtering by OS (LINUX): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_win32.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_mac64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux32.zip]
10:02:06.697 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs after filtering by OS (LINUX): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux32.zip]
10:02:06.697 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs before filtering by architecture (64): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux32.zip]
10:02:06.697 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs after filtering by architecture (64): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip]
10:02:06.698 TRACE[main] [io.github.bonigarcia.wdm.Downloader] getTarget 2.33 https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip
10:02:06.699 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Target path /home/jenkins/.m2/repository/webdriver
10:02:06.699 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Pre-download. target=/home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip, version=2.33
10:02:06.699 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Target file for URL https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip version 2.33 = /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip
10:02:06.706 INFO [main] [ChromeDriverProvider] Chromedriver path [null]
```
kaboom

**Comment by dimkin-eu** [comment]
just a hint
- `.forceCache()` makes things much better

**Comment by bonigarcia** [comment]
@dimkin-eu I committed a mechanism that (I think) is going to solve the problem. Please do the following:

* Use development version of wdm (`2.1.1-SNAPSHOT`). To do that, you will need to include the following repository in your project. In Maven:

```
    <repositories>
        <repository>
            <id>sonatype-snapshots</id>
            <url>http://oss.sonatype.org/content/repositories/snapshots</url>
            <layout>default</layout>
            <releases>
                <enabled>false</enabled>
            </releases>
            <snapshots>
                <enabled>true</enabled>
            </snapshots>
        </repository>
    </repositories>
```

* Clean your wdm cache (in your case located in `/home/jenkins/.m2/repository/webdriver`)
* Remove the `.forceCache()` option
* Launch again your parallel tests

Please let me know if the problem is gone now.

**Comment by bonigarcia** [comment]
@dimkin-eu I have uploaded a new version of wdm `2.1.1-SNAPSHOT`. Could you please repeat the experiment using the new artifact?

**Comment by dimkin-eu** [comment]
```
Downloaded: https://rms...com/public/io/github/bonigarcia/webdrivermanager/2.1.1-SNAPSHOT/webdrivermanager-2.1.1-20180312.094001-9.jar (61 KB at 18.6 KB/sec)
```

```
09:57:46.995 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key os.name not found, using default value
09:57:47.003 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.proxy not found, using default value
09:57:47.009 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.proxyUser not found, using default value
09:57:47.010 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.proxyPass not found, using default value
09:57:47.537 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] >> Managing chromedriver arch=64 version=2.33 getLatest=false cache=false
09:57:47.538 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Target path /home/jenkins/.m2/repository/webdriver
09:57:47.538 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking if chromedriver 2.33 (64 bits) exists in cache /home/jenkins/.m2/repository/webdriver
09:57:47.554 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/geckodriver/linux64/0.16.1/geckodriver
09:57:47.555 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver
09:57:47.555 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.30/chromedriver
09:57:47.555 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.29/chromedriver
09:57:47.555 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.28/chromedriver
09:57:47.556 INFO [main] [io.github.bonigarcia.wdm.WebDriverManager] Reading https://chromedriver.storage.googleapis.com/ to seek chromedriver
09:57:48.139 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] All URLs: [https://chromedriver.storage.googleapis.com/2.0/chromedriver_linux32.zip, 
```
```
09:57:48.140 DEBUG[main] [io.github.bonigarcia.wdm.WebDriverManager] Using chromedriver 2.33
09:57:48.140 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Candidate URLs: [https://chromedriver.storage.googleapis.com/2.33/chromedriver_win32.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_mac64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux32.zip]
09:57:48.142 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs before filtering by OS (LINUX): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_win32.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_mac64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux32.zip]
09:57:48.143 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs after filtering by OS (LINUX): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux32.zip]
09:57:48.143 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs before filtering by architecture (64): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux32.zip]
09:57:48.148 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs after filtering by architecture (64): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip]
09:57:48.149 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.ignoreVersions not found, using default value
09:57:48.150 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.ignoreVersions not found, using default value
09:57:48.150 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs before filtering by ignored versions ([]): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip]
09:57:48.150 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs after filtering by ignored versions ([]): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip]
09:57:48.151 TRACE[main] [io.github.bonigarcia.wdm.Downloader] getTarget 2.33 https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip
09:57:48.152 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Target path /home/jenkins/.m2/repository/webdriver
09:57:48.152 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Pre-download. target=/home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip, version=2.33
09:57:48.152 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Target file for URL https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip version 2.33 = /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip
09:57:48.153 TRACE[main] [io.github.bonigarcia.wdm.Downloader] [chromedriver] does not exist in cache
09:57:48.153 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Downloading https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip to /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip (true)
09:57:48.153 INFO [main] [io.github.bonigarcia.wdm.Downloader] Downloading https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip to /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip
09:57:48.156 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Using temporal file /tmp/2096667034914662855/chromedriver_linux64.zip
09:57:48.600 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Deleting file /tmp/2096667034914662855
09:57:48.601 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Compressed file /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip
09:57:48.601 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Unzipping chromedriver (size: 8950080 KB, compressed size: 4091529 KB)
09:57:48.602 DEBUG[main] [io.github.bonigarcia.wdm.Downloader] /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver already exists
09:57:48.603 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Deleting file /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip
09:57:48.608 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Setting file /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip as executable
09:57:48.608 WARN [main] [io.github.bonigarcia.wdm.Downloader] Error setting file /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip as executable
09:57:48.608 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Resulting binary file /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip
09:57:48.609 INFO [main] [io.github.bonigarcia.wdm.WebDriverManager] Exporting webdriver.chrome.driver as /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip
09:57:48.618 INFO [main] [ChromeDriverProvider] Chromedriver path [/home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip]
```
```
09:57:48.675 INFO [main] [ChromeDriverProvider] Exception thrown by chrome driver initialization
java.util.concurrent.ExecutionException: java.lang.IllegalStateException: The driver executable does not exist: /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip
```

**Comment by bonigarcia** [comment]
@dimkin-eu Thanks for helping in the debugging of this issue (unfortunately I am not able to reproduce it). I have committed a new version of the SNAPSHOT artifact.  Please repeat the test, let see if this time everything works fine.

**Comment by dimkin-eu** [comment]
@bonigarcia , no problems :) Locally I also not able to reproduce, only with jenkins matrix jobs
now it sets wrong webdriver version as exec (2.33 > 2.30)
```
Downloading: https://rms...com/public/io/github/bonigarcia/webdrivermanager/2.1.1-SNAPSHOT/webdrivermanager-2.1.1-20180312.103930-11.pom
```
```
10:48:24.335 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key os.name not found, using default value
10:48:24.348 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.proxy not found, using default value
10:48:24.355 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.proxyUser not found, using default value
10:48:24.356 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.proxyPass not found, using default value
10:48:24.797 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] >> Managing chromedriver arch=64 version=2.33 getLatest=false cache=false
10:48:24.798 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Target path /home/jenkins/.m2/repository/webdriver
10:48:24.798 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking if chromedriver 2.33 (64 bits) exists in cache /home/jenkins/.m2/repository/webdriver
10:48:24.811 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/geckodriver/linux64/0.16.1/geckodriver
10:48:24.811 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver
10:48:24.811 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.30/chromedriver
10:48:24.811 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.29/chromedriver
10:48:24.811 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.28/chromedriver
10:48:24.812 INFO [main] [io.github.bonigarcia.wdm.WebDriverManager] Reading https://chromedriver.storage.googleapis.com/ to seek chromedriver
10:48:25.536 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] All URLs: [https://chromedriver.storage.googleapis.com/2.0/chromedriver_linux32.zip, 
```
```
10:48:25.538 DEBUG[main] [io.github.bonigarcia.wdm.WebDriverManager] Using chromedriver 2.33
10:48:25.538 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Candidate URLs: [https://chromedriver.storage.googleapis.com/2.33/chromedriver_win32.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_mac64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux32.zip]
10:48:25.540 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs before filtering by OS (LINUX): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_win32.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_mac64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux32.zip]
10:48:25.540 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs after filtering by OS (LINUX): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux32.zip]
10:48:25.540 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs before filtering by architecture (64): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux32.zip]
10:48:25.541 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs after filtering by architecture (64): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip]
10:48:25.542 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.ignoreVersions not found, using default value
10:48:25.543 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.ignoreVersions not found, using default value
10:48:25.543 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs before filtering by ignored versions ([]): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip]
10:48:25.543 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs after filtering by ignored versions ([]): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip]
10:48:25.545 TRACE[main] [io.github.bonigarcia.wdm.Downloader] getTarget 2.33 https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip
10:48:25.546 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Target path /home/jenkins/.m2/repository/webdriver
10:48:25.546 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Pre-download. target=/home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip, version=2.33
10:48:25.546 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Target file for URL https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip version 2.33 = /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip
10:48:25.546 TRACE[main] [io.github.bonigarcia.wdm.Downloader] [chromedriver] does not exist in cache
10:48:25.547 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Downloading https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip to /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip (true)
10:48:25.547 INFO [main] [io.github.bonigarcia.wdm.Downloader] Downloading https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip to folder /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33
10:48:25.550 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Using temporal file /tmp/8197588714217171244/chromedriver_linux64.zip
10:48:26.129 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Compressed file /tmp/8197588714217171244/chromedriver_linux64.zip
10:48:26.130 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Unzipping chromedriver (size: 8950080 KB, compressed size: 4091529 KB)
10:48:26.221 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Renaming file from /tmp/8197588714217171244/c36f0f2e-384a-4605-966a-3af3a7f04781 to /tmp/8197588714217171244/chromedriver
10:48:26.222 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Setting file /tmp/8197588714217171244/chromedriver as executable
10:48:26.222 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Deleting file /tmp/8197588714217171244/chromedriver_linux64.zip
10:48:26.223 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Found binary in post-download: /tmp/8197588714217171244/chromedriver
10:48:26.223 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Setting file /tmp/8197588714217171244/chromedriver as executable
10:48:26.224 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Resulting binary file /tmp/8197588714217171244/chromedriver
10:48:26.230 WARN [main] [io.github.bonigarcia.wdm.WebDriverManager] There was an error managing chromedriver 2.33 (Destination '/home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver' already exists) ... trying again forcing to use cache
org.apache.commons.io.FileExistsException: Destination '/home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver' already exists
	at org.apache.commons.io.FileUtils.moveFile(FileUtils.java:2986) ~[commons-io-2.5.jar:2.5]
	at org.apache.commons.io.FileUtils.moveFileToDirectory(FileUtils.java:3033) ~[commons-io-2.5.jar:2.5]
	at io.github.bonigarcia.wdm.Downloader.downloadAndExtract(Downloader.java:120) ~[webdrivermanager-2.1.1-SNAPSHOT.jar:na]
	at io.github.bonigarcia.wdm.Downloader.download(Downloader.java:94) ~[webdrivermanager-2.1.1-SNAPSHOT.jar:na]
	at io.github.bonigarcia.wdm.WebDriverManager.downloadCandidateUrls(WebDriverManager.java:440) [webdrivermanager-2.1.1-SNAPSHOT.jar:na]
	at io.github.bonigarcia.wdm.WebDriverManager.manage(WebDriverManager.java:407) [webdrivermanager-2.1.1-SNAPSHOT.jar:na]
	at io.github.bonigarcia.wdm.WebDriverManager.setup(WebDriverManager.java:215) [webdrivermanager-2.1.1-SNAPSHOT.jar:na]

.....
10:48:26.232 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.proxy not found, using default value
10:48:26.232 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.proxyUser not found, using default value
10:48:26.232 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.proxyPass not found, using default value
10:48:26.235 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] >> Managing chromedriver arch=64 version=2.33 getLatest=false cache=true
10:48:26.235 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Target path /home/jenkins/.m2/repository/webdriver
10:48:26.235 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking if chromedriver exists in cache /home/jenkins/.m2/repository/webdriver
10:48:26.235 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/geckodriver/linux64/0.16.1/geckodriver
10:48:26.235 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver
10:48:26.236 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.30/chromedriver
10:48:26.236 INFO [main] [io.github.bonigarcia.wdm.WebDriverManager] Found chromedriver in cache: /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.30/chromedriver 
10:48:26.236 DEBUG[main] [io.github.bonigarcia.wdm.WebDriverManager] Driver for chromedriver 2.33 found in cache /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.30/chromedriver
10:48:26.236 INFO [main] [io.github.bonigarcia.wdm.WebDriverManager] Exporting webdriver.chrome.driver as /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.30/chromedriver
```

**Comment by bonigarcia** [comment]
@dimkin-eu I have committed a new change to avoid that problem... hopefully this time is the good one. Please repeat the test with the latest SNAPSHOT.

**Comment by dimkin-eu** [comment]
Wrong version again due IO - remember, same .m2 could be used by all parallel jobs

```
Downloaded: https://rms...com/public/io/github/bonigarcia/webdrivermanager/2.1.1-SNAPSHOT/webdrivermanager-2.1.1-20180312.112013-13.jar (61 KB at 15.3 KB/sec)
```

```
11:24:33.600 DEBUG[main] [io.github.bonigarcia.wdm.WebDriverManager] Using chromedriver 2.33
11:24:33.600 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Candidate URLs: [https://chromedriver.storage.googleapis.com/2.33/chromedriver_win32.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_mac64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux32.zip]
11:24:33.602 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs before filtering by OS (LINUX): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_win32.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_mac64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux32.zip]
11:24:33.602 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs after filtering by OS (LINUX): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux32.zip]
11:24:33.602 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs before filtering by architecture (64): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip, https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux32.zip]
11:24:33.605 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs after filtering by architecture (64): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip]
11:24:33.606 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.ignoreVersions not found, using default value
11:24:33.612 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.ignoreVersions not found, using default value
11:24:33.612 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs before filtering by ignored versions ([]): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip]
11:24:33.612 TRACE[main] [io.github.bonigarcia.wdm.UrlFilter] URLs after filtering by ignored versions ([]): [https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip]
11:24:33.612 TRACE[main] [io.github.bonigarcia.wdm.Downloader] getTarget 2.33 https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip
11:24:33.613 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Target path /home/jenkins/.m2/repository/webdriver
11:24:33.614 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Pre-download. target=/home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip, version=2.33
11:24:33.614 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Target file for URL https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip version 2.33 = /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip
11:24:33.614 TRACE[main] [io.github.bonigarcia.wdm.Downloader] [chromedriver] does not exist in cache
11:24:33.614 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Downloading https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip to /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip (true)
11:24:33.614 INFO [main] [io.github.bonigarcia.wdm.Downloader] Downloading https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip to folder /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33
11:24:33.616 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Using temporal file /tmp/6008284032998933936/chromedriver_linux64.zip
11:24:34.041 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Compressed file /tmp/6008284032998933936/chromedriver_linux64.zip
11:24:34.042 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Unzipping chromedriver (size: 8950080 KB, compressed size: 4091529 KB)
11:24:34.133 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Renaming file from /tmp/6008284032998933936/acb951f5-9140-414a-9dc0-08ec235f9c7c to /tmp/6008284032998933936/chromedriver
11:24:34.133 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Setting file /tmp/6008284032998933936/chromedriver as executable
11:24:34.133 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Deleting file /tmp/6008284032998933936/chromedriver_linux64.zip
11:24:34.135 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Found binary in post-download: /tmp/6008284032998933936/chromedriver
11:24:34.135 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Setting file /tmp/6008284032998933936/chromedriver as executable
11:24:34.135 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Resulting binary file /tmp/6008284032998933936/chromedriver
11:24:34.135 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Deleting file /tmp/6008284032998933936
11:24:34.146 WARN [main] [io.github.bonigarcia.wdm.WebDriverManager] There was an error managing chromedriver 2.33 (java.nio.file.DirectoryNotEmptyException: /tmp/6008284032998933936) ... trying again forcing to use cache
io.github.bonigarcia.wdm.WebDriverManagerException: java.nio.file.DirectoryNotEmptyException: /tmp/6008284032998933936
	at io.github.bonigarcia.wdm.Downloader.deleteFile(Downloader.java:346) ~[webdrivermanager-2.1.1-SNAPSHOT.jar:na]
	at io.github.bonigarcia.wdm.Downloader.downloadAndExtract(Downloader.java:124) ~[webdrivermanager-2.1.1-SNAPSHOT.jar:na]
	at io.github.bonigarcia.wdm.Downloader.download(Downloader.java:94) ~[webdrivermanager-2.1.1-SNAPSHOT.jar:na]
	at io.github.bonigarcia.wdm.WebDriverManager.downloadCandidateUrls(WebDriverManager.java:440) [webdrivermanager-2.1.1-SNAPSHOT.jar:na]
	at io.github.bonigarcia.wdm.WebDriverManager.manage(WebDriverManager.java:407) [webdrivermanager-2.1.1-SNAPSHOT.jar:na]
	at io.github.bonigarcia.wdm.WebDriverManager.setup(WebDriverManager.java:215) [webdrivermanager-2.1.1-SNAPSHOT.jar:na]
	
Caused by: java.nio.file.DirectoryNotEmptyException: /tmp/6008284032998933936
	at sun.nio.fs.UnixFileSystemProvider.implDelete(UnixFileSystemProvider.java:242) ~[na:1.8.0_144]
	at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103) ~[na:1.8.0_144]
	at java.nio.file.Files.delete(Files.java:1126) ~[na:1.8.0_144]
	at io.github.bonigarcia.wdm.Downloader.deleteFile(Downloader.java:344) ~[webdrivermanager-2.1.1-SNAPSHOT.jar:na]
	... 46 common frames omitted
11:24:34.147 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.proxy not found, using default value
11:24:34.148 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.proxyUser not found, using default value
11:24:34.148 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.proxyPass not found, using default value
11:24:34.150 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] >> Managing chromedriver arch=64 version=2.33 getLatest=false cache=true
11:24:34.151 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Target path /home/jenkins/.m2/repository/webdriver
11:24:34.151 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking if chromedriver exists in cache /home/jenkins/.m2/repository/webdriver
11:24:34.151 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/geckodriver/linux64/0.16.1/geckodriver
11:24:34.151 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver
11:24:34.151 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.30/chromedriver
11:24:34.152 INFO [main] [io.github.bonigarcia.wdm.WebDriverManager] Found chromedriver in cache: /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.30/chromedriver 
11:24:34.152 DEBUG[main] [io.github.bonigarcia.wdm.WebDriverManager] Driver for chromedriver 2.33 found in cache /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.30/chromedriver
11:24:34.152 INFO [main] [io.github.bonigarcia.wdm.WebDriverManager] Exporting webdriver.chrome.driver as /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.30/chromedriver
11:24:34.156 INFO [main] [ChromeDriverProvider] Chromedriver path [/home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.30/chromedriver]
```

**Comment by bonigarcia** [comment]
Sorry, my mistake. I have solved it. Please try again...

**Comment by dimkin-eu** [comment]
imho, there is no point to +x tmp file, only destination one
```
11:50:35.053 TRACE[main] [io.github.bonigarcia.wdm.Downloader] [chromedriver] does not exist in cache
11:50:35.053 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Downloading https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip to /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver_linux64.zip (true)
11:50:35.053 INFO [main] [io.github.bonigarcia.wdm.Downloader] Downloading https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip to folder /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33
11:50:35.055 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Using temporal file /tmp/5320323841482929240/chromedriver_linux64.zip
11:50:35.480 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Compressed file /tmp/5320323841482929240/chromedriver_linux64.zip
11:50:35.480 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Unzipping chromedriver (size: 8950080 KB, compressed size: 4091529 KB)
11:50:35.563 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Renaming file from /tmp/5320323841482929240/488e978e-8519-44b8-9055-3f0a6346972d to /tmp/5320323841482929240/chromedriver
11:50:35.564 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Setting file /tmp/5320323841482929240/chromedriver as executable
11:50:35.565 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Deleting file /tmp/5320323841482929240/chromedriver_linux64.zip
11:50:35.566 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Found binary in post-download: /tmp/5320323841482929240/chromedriver
11:50:35.566 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Setting file /tmp/5320323841482929240/chromedriver as executable
11:50:35.567 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Resulting binary file /tmp/5320323841482929240/chromedriver
11:50:35.567 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Deleting folder /tmp/5320323841482929240
11:50:35.573 INFO [main] [io.github.bonigarcia.wdm.Downloader] Binary driver after extraction /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver
11:50:35.574 INFO [main] [io.github.bonigarcia.wdm.WebDriverManager] Exporting webdriver.chrome.driver as /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver
11:50:35.580 INFO [main] [ChromeDriverProvider] Chromedriver path [/home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver]
11:50:35.632 INFO [main] [ChromeDriverProvider] Exception thrown by chrome driver initialization
java.util.concurrent.ExecutionException: java.lang.IllegalStateException: The driver is not executable: /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)```

**Comment by bonigarcia** [comment]
Grrr yes, that was another problem on the current logic (I have changed the way in which files are uncompressed and moved to the target folder). Let's see this time... please try again.

**Comment by dimkin-eu** [comment]
some logical bomb appeared - previous file wasnt executable, so this will not make it executable as well :)
```         
if (!resultingBinary.exists()) {
              moveFileToDirectory(extractedFile, targetFolder, true);
              moveFileToDirectory(extractedFile, targetFolder, true);
             setFileExecutable(resultingBinary);
          }		   
       }
```
```
12:08:50.706 INFO [main] [io.github.bonigarcia.wdm.Downloader] Downloading https://chromedriver.storage.googleapis.com/2.33/chromedriver_linux64.zip to folder /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33
12:08:50.708 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Using temporal file /tmp/1027032928799522766/chromedriver_linux64.zip
12:08:51.135 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Compressed file /tmp/1027032928799522766/chromedriver_linux64.zip
12:08:51.136 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Unzipping chromedriver (size: 8950080 KB, compressed size: 4091529 KB)
12:08:51.221 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Renaming file from /tmp/1027032928799522766/1556598b-a267-4ed0-91f0-55f1ed03e410 to /tmp/1027032928799522766/chromedriver
12:08:51.221 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Setting file /tmp/1027032928799522766/chromedriver as executable
12:08:51.221 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Deleting file /tmp/1027032928799522766/chromedriver_linux64.zip
12:08:51.223 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Found binary in post-download: /tmp/1027032928799522766/chromedriver
12:08:51.223 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Resulting binary file /tmp/1027032928799522766/chromedriver
12:08:51.223 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Deleting folder /tmp/1027032928799522766
12:08:51.231 INFO [main] [io.github.bonigarcia.wdm.Downloader] Binary driver after extraction /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver
12:08:51.232 INFO [main] [io.github.bonigarcia.wdm.WebDriverManager] Exporting webdriver.chrome.driver as /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver
12:08:51.239 INFO [main] [ChromeDriverProvider] Chromedriver path [/home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver]
12:08:51.328 INFO [main] [ChromeDriverProvider] Exception thrown by chrome driver initialization
java.util.concurrent.ExecutionException: java.lang.IllegalStateException: The driver is not executable: /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver
```

**Comment by bonigarcia** [comment]
Well... I think we are almost there ;)

Please @dimkin-eu, test one more time.

**Comment by dimkin-eu** [comment]
@bonigarcia , now something strange - there is no log from
`log.trace("Setting file {} as executable", file);`
and then
```
13:13:46.738 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking if chromedriver 2.33 (64 bits) exists in cache /home/jenkins/.m2/repository/webdriver
13:13:46.747 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/geckodriver/linux64/0.16.1/geckodriver
13:13:46.747 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver
13:13:46.747 DEBUG[main] [io.github.bonigarcia.wdm.WebDriverManager] Found 2.33 chromedriver (64 bits) in cache: /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver
13:13:46.748 DEBUG[main] [io.github.bonigarcia.wdm.WebDriverManager] Driver for chromedriver 2.33 found in cache /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver
13:13:46.748 INFO [main] [io.github.bonigarcia.wdm.WebDriverManager] Exporting webdriver.chrome.driver as /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver
13:13:46.752 INFO [main] [ChromeDriverProvider] Chromedriver path [/home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver]

Mar 12, 2018 1:14:06 PM org.openqa.selenium.os.OsProcess checkForError
SEVERE: org.apache.commons.exec.ExecuteException: Process exited with an error: 139 (Exit value: 139)
13:14:06.917 INFO [main] [ChromeDriverProvider] Exception thrown by chrome driver initialization
java.util.concurrent.ExecutionException: org.openqa.selenium.WebDriverException: Timed out waiting for driver server to start.
```

**Comment by dimkin-eu** [comment]
Something wrong with extraction (previously ?) - right size is ~8MB
for failed
```
+ ls -la /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/
total 1352
drwxrwxr-x 2 jenkins jenkins    4096 Mar 12 10:01 .
drwxrwxr-x 6 jenkins jenkins    4096 Oct 10 10:10 ..
-rwxrw-r-- 1 jenkins jenkins 1376159 Oct 10 10:10 chromedriver
```

for successful
```
+ ls -la /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/
total 8752
drwxrwxr-x 2 jenkins jenkins    4096 Oct 10 10:11 .
drwxrwxr-x 7 jenkins jenkins    4096 Oct 10 10:11 ..
-rwxrw-r-- 1 jenkins jenkins 8950080 Oct 10 10:11 chromedriver
```

**Comment by dimkin-eu** [comment]
looks like best solution - to purge .m2
but if `use cache` means `no download`, then driver must be redownloaded and reexported

**Comment by bonigarcia** [comment]
Maybe your cache were corrupted in a previous test. By default, wdm tries to use the cached file. To force the download to start there is the option `.forceDownload()`, as follows:

`WebDriverManager.chromedriver().forceDownload().version("2.33").setup();`

The option `.useCache()` tries to use latest cached version without asking to the online repository which one is the latest.

All in all... are your parallel tests working?

Just in case you want to test again, I have uploaded a new SNAPSHOT with an slight change (now the set executable logic is done at the end only if required).

**Comment by dimkin-eu** [comment]
`WebDriverManager.chromedriver().forceDownload().version("2.33").setup();`
this didnt helped - corrupted file still there. Also option not seen in log, could be nice to have for debugging

```
14:19:20.496 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key os.name not found, using default value
14:19:20.507 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.proxy not found, using default value
14:19:20.510 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.proxyUser not found, using default value
14:19:20.510 TRACE[main] [io.github.bonigarcia.wdm.Config] Property key wdm.proxyPass not found, using default value
14:19:20.898 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] >> Managing chromedriver arch=64 version=2.33 getLatest=false cache=false
14:19:20.899 TRACE[main] [io.github.bonigarcia.wdm.Downloader] Target path /home/jenkins/.m2/repository/webdriver
14:19:20.899 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking if chromedriver 2.33 (64 bits) exists in cache /home/jenkins/.m2/repository/webdriver
14:19:20.910 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/geckodriver/linux64/0.16.1/geckodriver
14:19:20.910 TRACE[main] [io.github.bonigarcia.wdm.WebDriverManager] Checking /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver
14:19:20.910 DEBUG[main] [io.github.bonigarcia.wdm.WebDriverManager] Found 2.33 chromedriver (64 bits) in cache: /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver
14:19:20.910 DEBUG[main] [io.github.bonigarcia.wdm.WebDriverManager] Driver for chromedriver 2.33 found in cache /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver
14:19:20.911 INFO [main] [io.github.bonigarcia.wdm.WebDriverManager] Exporting webdriver.chrome.driver as /home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver
14:19:20.918 INFO [main] [ChromeDriverProvider] Chromedriver path [/home/jenkins/.m2/repository/webdriver/chromedriver/linux64/2.33/chromedriver]
Mar 12, 2018 2:19:41 PM org.openqa.selenium.os.OsProcess checkForError
SEVERE: org.apache.commons.exec.ExecuteException: Process exited with an error: 139 (Exit value: 139)
14:19:41.150 INFO [main] [ChromeDriverProvider] Exception thrown by chrome driver initialization
java.util.concurrent.ExecutionException: org.openqa.selenium.WebDriverException: Timed out waiting for driver server to start.
```

**Comment by bonigarcia** [comment]
Now there were an additional problem with the `.forceDownload()` option. I have just committed a new patch and uploaded a new SNAPSHOT. Sorry for the inconveniences. Could you please test again?

**Comment by dimkin-eu** [comment]
@bonigarcia with latest (for this morning) build everything seems fine for me 👍 
Summoning @damonsmith to check ( make sure if you youse snapshot, if new release not done )

**Comment by bonigarcia** [comment]
Version 2.2.0 has been released, solving this issue. Thanks @damonsmith and @dimkin-eu for helping in the debug process.

**Comment by Jeevan-Palo** [comment]
Am getting this error
INFO: Trying to load configuration application.properties
[main] INFO io.github.bonigarcia.wdm.WebDriverManager - Using chromedriver 77.0.3865.40 (since Google Chrome 77 is installed in your machine)
[main] INFO io.github.bonigarcia.wdm.WebDriverManager - Exporting webdriver.chrome.driver as C:\Users\Jeevan\.m2\repository\webdriver\chromedriver\win32\77.0.3865.40\chromedriver.exe
